telegram/base.py

```diff
@@ -36,13 +36,20 @@
 class TelegramObject:
     """Base class for most Telegram objects."""
 
+    # type hints in __new__ are not read by mypy (https://github.com/python/mypy/issues/1021). As a
+    # workaround we can type hint instance variables in __new__ using a syntax defined in PEP 526 -
+    # https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations
+    if TYPE_CHECKING:
+        _id_attrs: Tuple[object, ...]
     # Adding slots reduces memory usage & allows for faster attribute access.
     # Only instance variables should be added to __slots__.
     __slots__ = ('_id_attrs',)
 
     def __new__(cls, *args: object, **kwargs: object) -> 'TelegramObject':  # pylint: disable=W0613
+        # We add _id_attrs in __new__ instead of __init__ since we want to add this to the slots
+        # w/o calling __init__ in all of the subclasses. This is what we also do in BaseFilter.
```

Comment:
having that explanation here won't help you very much while reading the source of `BaseFilter`, though 😄 😋 please add the note the as well - in fact the workaround for type checking may also be applicable?

Reply:
> in fact the workaround for type checking may also be applicable?

For `BaseFilter`? Yeah but we don't have any issues with type hints for it? Is it needed?

Reply:
Don't know … was just a thought since we have pretty much the same situation there …

Reply:
Just for completeness: I think the difference is that for filters we have the `name` setter + getter, so the child classes don't try to access `_name` directly